test/support/ray_api.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package support
+
+import (
+	"github.com/onsi/gomega"
+)
+
+func GetRayJobAPIDetails(t Test, rayClient RayClusterClient, jobID string) *RayJobDetailsResponse {
+	t.T().Helper()
+	return RayJobAPIDetails(t, rayClient, jobID)(t)
+}
+
+func WriteRayJobAPILogs(t Test, rayClient RayClusterClient, jobID string) {
```

Comment:
I'd suggest to remove the `GetRayJobLogs` and `WriteRayJobLogs` functions from the `ray.go` file, that rely on the API server proxy, and have them replaced by these one, so there are used in the tests upstream as well.

For it to work, we can install the NGINX ingress controller in KinD (The KinD configuration already contains the "ingress-ready=true" kubelet extra argument), and create an Ingress resource in the `TestMNISTRayJobMCADRayCluster` test.

Reply:
@astefanutti Created https://github.com/project-codeflare/codeflare-operator/issues/199